### PR TITLE
Add pull builder for Fetch and Batch

### PR DESCRIPTION
**Pull request by Jarema**

### CODE REVIEW

File: async-nats/src/jetstream/consumer/pull.rs [hunk 1]
```diff
@@ -205,26 +196,16 @@ impl Consumer<Config> {
...
-                batch,
-                expires,
-                no_wait: false,
-                idle_heartbeat: Duration::default(),
-                ..Default::default()
-            },
-            self,
-        )
-        .await
+    pub fn batch(&self) -> BatchBuilder {
```
Comment (caspervonb) [inline async-nats/src/jetstream/consumer/pull.rs:207]: So the distinction now, between fetch and batch is no_wait default which is not configurable?

Comment (Jarema) [reply async-nats/src/jetstream/consumer/pull.rs:208]: yes.

File: async-nats/src/jetstream/consumer/pull.rs [hunk 2]
```diff
@@ -52,15 +52,15 @@ impl Consumer<Config> {
...
+    /// let mut messages = consumer.messages().await?.take(100);
     /// while let Some(Ok(message)) = messages.next().await {
     ///   println!("got message {:?}", message);
     ///   message.ack().await?;
     /// }
     /// Ok(())
     /// # }
     /// ```
-    pub async fn stream(&self) -> Result<Stream<'_>, Error> {
+    pub async fn messages(&self) -> Result<Stream<'_>, Error> {
```
[the remark this answers is not in the document]
Comment (Jarema) [reply async-nats/src/jetstream/consumer/pull.rs:63]: yep, changed to that already.

File: async-nats/src/jetstream/consumer/pull.rs [hunk 3]
```diff
@@ -714,14 +695,143 @@ impl<'a> StreamBuilder<'a> {
...
+///     println!("message: {:?}", message);
+///     message.ack().await?;
+/// }
+/// # Ok(())
+/// # }
+pub struct FetchBuilder<'a> {
+    batch: usize,
+    max_bytes: usize,
+    hearbeat: Duration,
+    expires: usize,
```
[the remark this answers is not in the document]
Comment (caspervonb) [reply async-nats/src/jetstream/consumer/pull.rs:770]: Alright

**Review by caspervonb — APPROVED** [review]
lgtm

**Pull request merged**